Compatability with screen-dedicated tags

RandumbPurson opened the pull request:
I wanted to have some tags that were shared and some that were dedicated to each screen, and I sort of just installed this alongside the default config. I got an error and fixed it, so I figured I'd push this here so it would be easier for other people to get working if they want to.

In the `sharedtags.movetag` method, when sorting the tags, it tries to compare the attribute `.sharedtagindex` but its getting the tags from the screens, so it includes tags not created through `sharedtags.new` meaning that attribute often isn't set. I added a small helper function that just returns `sharedtagindex` if set (+ an arbitrary int to always place shared tags on the end), and `index` otherwise. With that change everything works, and for people just using this normally nothing changes.

Also I think my linter messed with the formatting, but I can go back and change it if you want.

RandumbPurson committed:
Fix compatability with basic tags + formatting

RandumbPurson committed:
Fix compatability with basic tags + formatting

RandumbPurson committed:
Merge branch 'v4.0' of https://github.com/RandumbPurson/awesome-sharedtags into v4.0

RandumbPurson committed:
Make default 0 for no standard user change

Drauthius commented on init.lua:
```diff
@@ -116,6 +116,15 @@ function sharedtags.new(def)
     return tags
 end
 
+function getidx(tag)
```
Should be marked `local`, and moved up to the other local function.

Drauthius commented on init.lua:
```diff
@@ -116,6 +116,15 @@ function sharedtags.new(def)
 
+function getidx(tag)
+    if tag.sharedtagindex then
+        -- Add arbitrarily large number to always be at the end
+        return tag.sharedtagindex + 0
```
I don't get this. How is zero large?

RandumbPurson replied:
Sorry, I forgot to change this back, I was just testing to see if an offset would affect normal users. This should be a number like 10, so that the shared tags always appear after normal tags.

RandumbPurson committed:
Make function local, add docstring, and fix offset

RandumbPurson commented:
I moved the `getidx` function and made it local as well as adding a comment to kind of explain its purpose. Also fixed the issues with incorrect offsets.

Drauthius commented:
Thanks for sharing your change and the fixes.
